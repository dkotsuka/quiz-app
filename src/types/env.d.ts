declare module '@env' {
    export const API_URL: string;
    export const API_TOKEN_URL: string;
    export const API_CATEGORY_URL: string;
}