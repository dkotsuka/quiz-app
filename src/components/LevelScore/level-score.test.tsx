import React from 'react'
import { LevelScore } from '.'
import { create } from 'react-test-renderer';
import { Score } from '../../screens/QuestionScreen/question-screen.types';

describe("LevelScore", () => {
    const score: Score = {
        easy: [1,2],
        medium: [3,4],
        hard: [5,6]
    }
    const easyScore = create(<LevelScore score={score} level="easy"/>)
    const mediumScore = create(<LevelScore score={score} level="medium"/>)
    const hardScore = create(<LevelScore score={score} level="hard"/>)

    it("maches snapshot", () => {
        expect(easyScore).toMatchSnapshot()
        expect(mediumScore).toMatchSnapshot()
        expect(hardScore).toMatchSnapshot()
    })

    it("counts correctly", () => {
        const correctText = easyScore.root.findByProps({testID:"level-score-text-correct"}).props
        const wrongText = easyScore.root.findByProps({testID:"level-score-text-wrong"}).props

        expect(correctText.children).toEqual(["Correct: ", 1])
        expect(wrongText.children).toEqual(["Wrong: ", 2])
    })

    it("counts correctly", () => {
        const correctText = mediumScore.root.findByProps({testID:"level-score-text-correct"}).props
        const wrongText = mediumScore.root.findByProps({testID:"level-score-text-wrong"}).props

        expect(correctText.children).toEqual(["Correct: ", 3])
        expect(wrongText.children).toEqual(["Wrong: ", 4])
    })

    it("counts correctly", () => {
        const correctText = hardScore.root.findByProps({testID:"level-score-text-correct"}).props
        const wrongText = hardScore.root.findByProps({testID:"level-score-text-wrong"}).props

        expect(correctText.children).toEqual(["Correct: ", 5])
        expect(wrongText.children).toEqual(["Wrong: ", 6])
    })
})