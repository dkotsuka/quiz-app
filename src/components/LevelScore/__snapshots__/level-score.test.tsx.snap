// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LevelScore maches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "flexDirection": "column",
        "justifyContent": "flex-start",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Roboto Bold",
          "fontSize": 20,
        },
        Object {
          "fontSize": 14,
        },
      ]
    }
  >
    Easy
  </Text>
  <View
    height={4}
    style={
      Array [
        Object {
          "height": 4,
        },
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Roboto",
          "fontSize": 14,
        },
        Object {
          "fontSize": 14,
        },
      ]
    }
    testID="level-score-text-correct"
  >
    Correct: 
    1
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Roboto",
          "fontSize": 14,
        },
        Object {
          "fontSize": 14,
        },
      ]
    }
    testID="level-score-text-wrong"
  >
    Wrong: 
    2
  </Text>
</View>
`;

exports[`LevelScore maches snapshot 2`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "flexDirection": "column",
        "justifyContent": "flex-start",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Roboto Bold",
          "fontSize": 20,
        },
        Object {
          "fontSize": 14,
        },
      ]
    }
  >
    Medium
  </Text>
  <View
    height={4}
    style={
      Array [
        Object {
          "height": 4,
        },
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Roboto",
          "fontSize": 14,
        },
        Object {
          "fontSize": 14,
        },
      ]
    }
    testID="level-score-text-correct"
  >
    Correct: 
    3
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Roboto",
          "fontSize": 14,
        },
        Object {
          "fontSize": 14,
        },
      ]
    }
    testID="level-score-text-wrong"
  >
    Wrong: 
    4
  </Text>
</View>
`;

exports[`LevelScore maches snapshot 3`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "flexDirection": "column",
        "justifyContent": "flex-start",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Roboto Bold",
          "fontSize": 20,
        },
        Object {
          "fontSize": 14,
        },
      ]
    }
  >
    Hard
  </Text>
  <View
    height={4}
    style={
      Array [
        Object {
          "height": 4,
        },
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Roboto",
          "fontSize": 14,
        },
        Object {
          "fontSize": 14,
        },
      ]
    }
    testID="level-score-text-correct"
  >
    Correct: 
    5
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Roboto",
          "fontSize": 14,
        },
        Object {
          "fontSize": 14,
        },
      ]
    }
    testID="level-score-text-wrong"
  >
    Wrong: 
    6
  </Text>
</View>
`;
